### PR TITLE
use shorter cache timeout

### PR DESCRIPTION
**Pull request by LesnyRumcajs**
**Summary of changes**
Changes introduced in this pull request:
- as @lemmih suggested, if the cache action doesn't complete within 2-3 minutes then it's most likely stalled and we would timeout anyway.
- continue the job even if the cache failed for whatever reason. If it happens, we will just rebuild.

**Reference issue to close (if applicable)**

Closes 

**Other information and links**

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -30,6 +31,7 @@ jobs:
         uses: actions/checkout@v3
       - name: Rust Cache
         uses: Swatinem/rust-cache@v2
+        timeout-minutes: $CACHE_TIMEOUT_MINUTES
```
Comment (hanabi1224) [inline .github/workflows/rust.yml:34]: I guess this has to be ${{ env.CACHE_TIMEOUT_MINUTES }}

File: .github/workflows/docs.yml [hunk 2]
```diff
@@ -31,6 +31,7 @@ jobs:
           fail: true
       - name: Rust Cache
         uses: Swatinem/rust-cache@v2
+        timeout-minutes: 5
```
Comment (lemmih) [inline .github/workflows/docs.yml:34]: `continue-on-error` here?

File: .github/workflows/rust.yml (hunk 1 above)
Comment (LesnyRumcajs) [reply .github/workflows/rust.yml]: It was a bit more complex, this piece of technology is somehow typed so there's a need to `fromJSON` https://docs.github.com/en/actions/learn-github-actions/expressions#fromjson
> Returns a JSON object or JSON data type for value. You can use this function to provide a JSON object as an evaluated expression or to convert environment variables from a string.

Thanks for looking into it anyway!

File: .github/workflows/docs.yml (hunk 2 above)
Comment (LesnyRumcajs) [reply .github/workflows/docs.yml:34]: good catch, I forgot to save the file :crying_cat_face:

**Review by hanabi1224 — APPROVED** [review]

**Review by lemmih — APPROVED** [review]

**Comment by lemmih** [comment]
Hurrah for @LeshyBot!

**Pull request merged**